File: problemas/iniciante/2152/2152.r
input <- file('stdin', 'r')

duasCasas <- function(numero) {
  if (numero < 10) {
    return(paste('0', toString(numero), sep=''))
  } else {
    return(toString(numero))
  }
}

n <- as.integer(readLines(input, n = 1))
for(i in 1:n) {
  entrada <- as.integer(strsplit(readLines(input, n = 1), ' ')[[1]])
  h <- entrada[1]
  m <- entrada[2]
  o <- entrada[3]

  if(o == 1) {
    write(paste(duasCasas(h), ':', duasCasas(m), ' - A porta abriu!', sep=''), '')
  } else {
    write(paste(duasCasas(h), ':', duasCasas(m), ' - A porta fechou!', sep=''), '')
  }
}

File: problemas/iniciante/2152/2152.lua
n = tonumber(io.read(), 10)
for _ = 1, n do
  entrada = {}
  for numero in io.read():gmatch("%S+") do
    entrada[#entrada + 1] = numero
  end
  h = tonumber(entrada[1], 10)
  m = tonumber(entrada[2], 10)
  o = tonumber(entrada[3], 10)

  if (o == 1) then
    print(string.format("%02d:%02d - A porta abriu!", h, m))
  else
    print(string.format("%02d:%02d - A porta fechou!", h, m))
  end
end
